pkg/crd/markers/patch_validation.go

```diff
@@ -64,3 +69,22 @@ func (m FeatureSetEnum) ApplyToSchema(schema *apiext.JSONSchemaProps) error {
 	schema.Enum = vals
 	return nil
 }
+
+type FeatureSetXValidation struct {
+	FeatureSetNames []string `marker:"featureSet"`
```

Comment:
Just a nit. Do you wanna keep the `featureSet` keyword instead of `featureSetNames`? Users will assume one can specify only a single featureset name and might actually produce:
```
// +openshift:validation:FeatureSetAwareXValidation:featureSet=CustomNoUpgrade,rule="has(oldSelf.coolNewField) ? has(self.coolNewField) : true",message="coolNewField may not be removed once set"
// +openshift:validation:FeatureSetAwareXValidation:featureSet=TechPreviewNoUpgrade,rule="has(oldSelf.coolNewField) ? has(self.coolNewField) : true",message="coolNewField may not be removed once set"
```
instead of combining both names into a single line.

Reply:
I've left it as `featureSet` so that it's consistent with the other feature set aware types we've created, they also have `featureSet` but allow multiple values